### PR TITLE
Add missing methods on IResult<I,O,E>

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/internal.rs [hunk 1]
```diff
@@ -46,25 +46,25 @@ pub enum IResult<I,O,E=u32> {
   Incomplete(Needed)
 }
 
-impl<I,O> IResult<I,O> {
+impl<I,O,E> IResult<I,O,E> {
   pub fn is_done(&self) -> bool {
-    match *self {
-      Done(_,_) => true,
-      _         => false
+    match self {
```
Comment (Geal) [inline src/internal.rs]: why change that? matching on `*self` is valid. It is even recommended by clippy

File: src/internal.rs [hunk 2]
```diff
@@ -46,25 +46,25 @@ pub enum IResult<I,O,E=u32> {
...
     }
   }
 
   pub fn is_err(&self) -> bool {
-    match *self {
-      Error(_) => true,
-      _        => false
+    match self {
+      &Error(_) => true,
+      _         => false
```
Comment (Geal) [inline src/internal.rs]: same comment

File: src/internal.rs [hunk 3]
```diff
@@ -46,25 +46,25 @@ pub enum IResult<I,O,E=u32> {
...
     }
   }
 
   pub fn is_incomplete(&self) -> bool {
-    match *self {
-      Incomplete(_) => true,
-      _             => false
+    match self {
+      &Incomplete(_) => true,
+      _              => false
```
Comment (Geal) [inline src/internal.rs]: same comment

File: src/internal.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/internal.rs]: Absolutely, mistake I made, my bad.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by Geal** [comment]
Thanks!

**Pull request merged**